core/modules/qdisp/QueryResource.cc

```diff
@@ -44,46 +44,53 @@
 #include "qdisp/JobStatus.h"
 #include "qdisp/QueryRequest.h"
 #include "qdisp/QueryResource.h"
-#include "qdisp/ResponseRequester.h"
 
 namespace lsst {
 namespace qserv {
 namespace qdisp {
 
+QueryResource::QueryResource(std::shared_ptr<JobQuery> jobQuery)
+      // this char* must live as long as this object, so copy it on heap
+    : Resource(::strdup(jobQuery->getDescription().resource().path().c_str())),
+      _xrdSsiSession(NULL), _jobQuery(jobQuery) {
```

Comment:
QueryResource needs the information in _jobQuery to make the QueryRequest (such as JobStatus). It's easier to keep that all in _jobQuery than passing the one pointer instead of passing and/or copying things. I think it is also clearer.